### PR TITLE
DHFPROD-6476: Step processor failure now reports correct job counts

### PR DESCRIPTION
**Pull request by rjrudin**
### Description

Had to remove the throwing of an exception from stopJob - I couldn't find any reason why that would be good. With it there, the test was failing because 4 failed items were being reported instead of 2. Specifically, a stepItemFailureListener was throwing this exception, after the 2 failed items had already been captured. So it seems unnecessary to throw an exception when stopJob is called but the jobId cannot be found. Also ran all of the flow tests and none of them failed with this change. 

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs [hunk 1]
```diff
@@ -395,22 +395,27 @@ class Flow {
...
+      contentArray.forEach(contentObject => {
+        if (contentObject.uri) {
+          this.writeQueue.push(contentObject);
+        } else {
+          this.datahub.debug.log({ type: 'error', message: `Couldn't add '${xdmp.toJsonString(contentObject)}' to the write queue due to missing uri.`});
+        }
+      });
+    }
 
     if (hook && !hook.runBefore) {
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs:420]: If a step has both processor and custom hook, should post commit hook run if processor fails ?

Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs:420]: I went back and forth on this. Because a custom hook is already able to run when items in a batch fail, I think it should be able to run when a processor fails too.

**Review by ryanjdew — APPROVED** [review]

**Review by srinathgit — APPROVED** [review]

**Review by akshaysonvane — APPROVED** [review]
